### PR TITLE
dns/bind: add SSHFP Record Type

### LINKED ISSUE #4119: os-bind: add SSHFP Record Type
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [X] When the request is meant for an existing plugin, I've added its name to the title.

**Is your feature request related to a problem? Please describe.**

Currently, the OPNsense BIND plugin does not support SSHFP records, which are important for securing SSH connections by verifying the SSH host keys through DNS. This lack of support creates an inconvenience as administrators have to manually handle SSH host key verification, increasing the risk of man-in-the-middle attacks.

**Describe the solution you'd like**

I would like to request the addition of SSHFP record support in the OPNsense BIND plugin. This would involve:

- Updating theuser interface for creating and managing SSHFP records besides the already supported types.
- Updating the BIND configuration templates to include SSHFP records.
- Ensuring the DNS zone files are correctly updated to reflect SSHFP records.

**Describe alternatives you've considered**

- Manual Addition: Manually adding SSHFP records directly to the BIND configuration files. This is not user-friendly and can lead to configuration errors. They get deleted after updating the zone.

- External Scripts: Using external scripts to automate the addition of SSHFP records. This requires additional maintenance and integration work.

- Third-party Tools: Leveraging third-party DNS management tools that support SSHFP records, but this adds complexity and may not integrate well with OPNsense.

**Additional context**

SSHFP records are a critical part of securing SSH communications and integrating this feature into the OPNsense BIND plugin will enhance the overall security posture of OPNsense deployments. This feature will be particularly beneficial for environments with a large number of SSH servers, where managing SSH host keys manually is impractical.

### PR DESCRIPTION
**Pull request by doktornotor**
Closes #4119

### CODE REVIEW

File: dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Record.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <model>
     <mount>//OPNsense/bind/record</mount>
     <description>BIND record configuration</description>
-    <version>1.0.1</version>
+    <version>1.0.2</version>
```
Comment (fichtner) [inline dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Record.xml:4]: Only need this when introducing new defaults or required fields

File: dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Record.xml [hunk 2]
```diff
@@ -40,6 +40,7 @@
                         <RP>RP</RP>
                         <RRSIG>RRSIG</RRSIG>
                         <SRV>SRV</SRV>
+                        <SRV>SSHFP</SRV>
```
Comment (fichtner) [inline dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Record.xml:43]: The XML tag name is the key so it can't be "SRV" again.

Comment (fichtner) [reply dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Record.xml:43]: (looks a bit silly, the text in the tag is the human readable and translatable label. In this case it's the same.

Comment (doktornotor) [reply dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Record.xml:43]: Well, I clearly need another bucket of ☕

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**